editors/code/src/commands/cargo_watch.ts

```diff
@@ -0,0 +1,137 @@
+import * as child_process from 'child_process';
+import * as path from 'path';
+import * as timers from 'timers';
+import * as vscode from 'vscode';
+import { StatusDisplay } from './watch_status';
+
+export class CargoWatchProvider {
+    private diagnosticCollection?: vscode.DiagnosticCollection;
+    private cargoProcess?: child_process.ChildProcess;
+    private outBuffer: string = '';
+    private statusDisplay?: StatusDisplay;
+
+    public activate(subscriptions: vscode.Disposable[]) {
+        subscriptions.push(this);
+        this.diagnosticCollection = vscode.languages.createDiagnosticCollection(
+            'rustc'
+        );
+
+        this.statusDisplay = new StatusDisplay(subscriptions);
+
+        // Start the cargo watch with json message
+        this.cargoProcess = child_process.spawn(
+            'cargo',
+            ['watch', '-x', '"check --message-format json"'],
+            {
+                // stdio: ['ignore', 'pipe', 'ignore'],
```

Comment:
this is for debugging, right?

Reply:
Yes, and i will add `rust-analyzer.trace.cargo-watch` config for enable/disable proper logging, do you think this is okay ?

Reply:
sgtm